src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -314,17 +318,60 @@ void WriteTypeMappings (List<TypeDefinition> types)
 			using (var gen = UseSharedRuntime
 				? new TypeNameMapGenerator (types, Log.LogDebugMessage)
 			        : new TypeNameMapGenerator (ResolvedAssemblies.Select (p => p.ItemSpec), Log.LogDebugMessage)) {
-				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), gen.WriteJavaToManaged);
-				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), gen.WriteManagedToJava);
+				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.jm"), "jm", gen.WriteJavaToManaged);
+				UpdateWhenChanged (Path.Combine (OutputDirectory, "typemap.mj"), "mj", gen.WriteManagedToJava);
 			}
 		}
 
-		void UpdateWhenChanged (string path, Action<Stream> generator)
+		void UpdateWhenChanged (string path, string type, Action<Stream> generator)
 		{
 			using (var stream = new MemoryStream ()) {
 				generator (stream);
 				MonoAndroidHelper.CopyIfStreamChanged (stream, path);
 			}
+
+			string dataFilePath = $"{path}.inc";
+			using (var stream = new NativeAssemblyDataStream ()) {
+				generator (stream);
```

Comment:
Can we reuse the first `MemoryStream` in this method, so we don't call `generator (stream)` twice?

Reply:
We could, but the stream is closed by `MonoAndroidHelper.CopyIfStreamChanged (stream, path);`